Lib/asyncio/tasks.py

```diff
@@ -42,9 +42,19 @@ def all_tasks(loop=None):
     """Return a set of all tasks for the loop."""
     if loop is None:
         loop = events.get_running_loop()
-    # NB: set(_all_tasks) is required to protect
+    # NB: list(_all_tasks) is required to protect
     # from https://bugs.python.org/issue34970 bug
-    return {t for t in list(_all_tasks)
+    # NB: Have to repeat on RuntimeError, other thread
+    # can modify _all_tasks during list(_all_tasks) call
+    # https://bugs.python.org/issue36607
+    while True:
+        try:
+            tasks = list(_all_tasks)
+        except RuntimeError:
+            pass
```

Comment:
I don't like unchecked loops like that. There must be a retry count or something.

Reply:
The number of iterations is the question.
What is good enough to prevent a collision? `10`? `100`? `1000`?
I think the number depends on count of threads and a number of tasks inside every processed loop.

Reply:
I'd say 1000 should be enough.  It's high enough to give it enough attempts and it's low enough to abort early if something weird is going on.